### PR TITLE
Potential fix related to Irrigation Tank

### LINKED ISSUE #1392: Maybe water tank will cause game crash randomly (Ticking block entity)
## Quick description

Just walk around to some areas or some time will cause crash .

## Related issues and searches

* https://github.com/AgriCraft/AgriCraft/issues?q=is%3Aissue%20EXAMPLE

## Mods and versions used

* Minecraft: 1.16.5
* AgriCraft: 3.0.4
* InfinityLib: 2.0.5
* AgriCraft OptiFine Compat: 0.0.1 
* OptiFine: 1.16.5_HD_U_G8
* Pretty large modpack about 220 mods

## Steps to replicate the issue

1. walk around to some areas (cause game crash many times in same area or some time).

## Expectations and explanations

My friend is building a water tank at same time.
But won't crash looking water tank direction.
My friend did not crash when building the water tank, but other players nearby cause crash randomly(some player not crash).
## Your environment

* How are you playing?
  * Operating System: Windows 10
  * Launcher: Default
  * Modpack: Private
* What kind of server:
  * [ ] Singleplayer
  * [X] Multiplayer
* Game Modes:
  * [ ] Survival
  * [X] Creative
* Save File:
  * [X] Brand new save
  * [X] Only an existing save

## Links to crash logs, stack traces, etc

* https://gist.github.com/catmunyi/a63496d51288329086a7988031da46d8

### PR DESCRIPTION
**Pull request by ArthixFR**
Hi, this pull may fix #1392. Tested in singleplayer and LAN, it seems to works well.

### CODE REVIEW

**Comment by InfinityRaider** [comment]
Thanks!
I'll have to look at this in more detail before I merge it though.

File: src/main/java/com/infinityraider/agricraft/content/irrigation/TileEntityIrrigationTank.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public BlockPos getMultiBlockMax() {
 
     public boolean isMultiBlockOrigin() {
         BlockPos origin = this.getMultiBlockMin();
-        return DEFAULT.equals(origin) || this.getPos().equals(origin);
+        return (this.origin != null && this.origin.getTile(world) == this) || DEFAULT.equals(origin) || this.getPos().equals(origin);
```
Comment (InfinityRaider) [inline src/main/java/com/infinityraider/agricraft/content/irrigation/TileEntityIrrigationTank.java:78]: Might want to move this to the end of the or chain as getting a tile entity from the world is the more expensive call.

File: src/main/java/com/infinityraider/agricraft/content/irrigation/TileEntityIrrigationTank.java [hunk 2]
```diff
@@ -307,7 +308,7 @@ public void unFormMultiBlock() {
                     TileEntity tile = this.getWorld().getTileEntity(pos);
                     if(tile instanceof TileEntityIrrigationTank) {
                         TileEntityIrrigationTank tank = (TileEntityIrrigationTank) tile;
-                        tank.origin = null;
+                        tank.origin = this.getWorld().isRemote() ? new TileReference<>(tank.getPos(), TANK_GETTER) : null;
```
Comment (InfinityRaider) [inline src/main/java/com/infinityraider/agricraft/content/irrigation/TileEntityIrrigationTank.java:311]: I like this, less objects being potentially null avoids NPE"s

File: src/main/java/com/infinityraider/agricraft/content/irrigation/TileEntityIrrigationTank.java (hunk 1 above)
Comment (ArthixFR) [reply src/main/java/com/infinityraider/agricraft/content/irrigation/TileEntityIrrigationTank.java]: You're right, done !

**Pull request merged**